core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -882,14 +882,23 @@ class AutoTuner(
     val lookup = "spark.sql.shuffle.partitions"
     var shufflePartitions =
       getPropertyValue(lookup).getOrElse(DEF_SHUFFLE_PARTITIONS).toInt
+    val shuffleStagesWithPosSpilling = appInfoProvider.getShuffleStagesWithPosSpilling
+
     // TODO: Need to look at other metrics for GPU spills (DEBUG mode), and batch sizes metric
     if (isCalculationEnabled(lookup)) {
-      val totalSpilledMetrics = appInfoProvider.getSpilledMetrics.sum
-      if (totalSpilledMetrics > 0) {
-        shufflePartitions *= DEF_SHUFFLE_PARTITION_MULTIPLIER
-        // Could be memory instead of partitions
-        appendOptionalComment(lookup,
-          s"'$lookup' should be increased since spilling occurred.")
+      if (shuffleStagesWithPosSpilling.nonEmpty) {
+        val shuffleSkewStages = appInfoProvider.getShuffleSkewStages
+        if (shuffleSkewStages.exists(id => shuffleStagesWithPosSpilling.contains(id))) {
```

Comment:
This check would make sure that the autotuner would not recommend increasing the shuffle partitions if there is atleast one stage with skewed data. Was wondering instead of checking for atleast one, can we still recommend increasing `number of shuffle pariitions` until a certain threshold (Ex: ratio of number of number of stages with skew to total number of stages with spills ).  
This can be a follow-on as in improvement if it takes more time to test.

Reply:
Interesting. What would be a good way to determine the threshold?

Reply:
Also, since both are lists of stage ids, can we use `intersect` for better performance?

```suggestion
        if (shuffleSkewStages.intersect(shuffleStagesWithPosSpilling).nonEmpty) {
```

Reply:
@nartal1 Thanks! I think this would require more discussion/testing. I can file a follow up issue to track this when we are ready to merge this PR.

Reply:
In theory `a.intersect(b)` should have complexity of `O(min(a, b))` and `a.exists(b.contains)` can have complexity of `O(a*b)`.  

Although `exists` can short circuit, the inner `contains` cause a full iteration of the list every time.

Reply:
P.S:`getShuffleStagesWithPosSpilling` already tries to find all the IDs. If we want a btetr performance then we will need to use another API that returns True/False if at least one exists. I did not want to request that change initially to keep the API standard for the first phase.

Back to the performance discussion here, In worst case `b.contains` is O(n) because `b` is a sequence. This is easily can be improved if we change the returnType of `getShuffleStagesWithPosSpilling` to `set` instead of sequence. this yields into 

`a.intersect(b)` the problem here that those are also sequences, which means that O(a*b). Even if we change it to Set, the output of the operation is the length of the intersection which is bigger than an operation that tries to answer "True/False"

Reply:
For the current approach, we need the full list of `getShuffleStagesWithPosSpilling`'s output to compare with `shuffleSkewStages` to check if they have any intersection. I will update `getShuffleStagesWithPosSpilling` and `getShuffleSkewStages` to return Set[Long] for better performance.